account_cutoff_accrual_sale/models/sale_order_line.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2018 Jacques-Etienne Baudoux (BCIM sprl) <je@bcim.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class SaleOrderLine(models.Model):
+    _name = "sale.order.line"
+    _inherit = ["sale.order.line", "order.line.cutoff.accrual.mixin"]
+
+    account_cutoff_line_ids = fields.One2many(
+        "account.cutoff.line",
+        "sale_line_id",
+        string="Account Cutoff Lines",
+        readonly=True,
+    )
+
+    is_cutoff_accrual_excluded = fields.Boolean(
+        compute="_compute_is_cutoff_accrual_excluded",
+        store=True,
+    )
+
+    @api.depends("order_id.force_invoiced")
+    def _compute_is_cutoff_accrual_excluded(self):
+        for rec in self:
+            rec.is_cutoff_accrual_excluded = rec.order_id.force_invoiced
```

Comment:
Why not only 'invoice_status' as force_invoiced will change it ?

Reply:
I trigger cutoff update on manual action on force invoiced flag. I don't want to trigger it each time a sale invoice status changes. And I don't know a good way to trigger some code execution on change of a stored computed field.

Reply:
> I trigger cutoff update on manual action on force invoiced flag. I don't want to trigger it each time a sale invoice status changes. And I don't know a good way to trigger some code execution on change of a stored computed field.

So, cutoff trigger does not occur at each time a SO line quantity is delivered and invoiced (can occur at several different times) ?

Reply:
* not at delivery as you cannot deliver in the past
* not at invoice creation (SO line qty_invoiced) but at invoice posting as this is when it is accounted

This makes me think that I cannot rely at all on the invoice status as this is changed at invoice creation and not when posted. So, I'll have to revert this change in the query domain.

Reply:
No, it's all good, I process also the draft invoices at https://github.com/OCA/account-closing/pull/271/files#diff-df279c387c73fffb2fef47c05b5c5dc189e1ddb956a175de21a5a4f69300af55R64